### DISCUSSION TITLE
How to send HttpOnly cookies?

### DISCUSSION THREAD

**Discussion by christophediprima** [enisdenjo/graphql-ws#497] [Q&A] [answered]
I am currently trying to migrate from subscriptions-transport-ws and could not find a way to send the HttpOnly cookies.

The GraphQL server I am trying to hit is protected by a Kong gateway that uses HttpOnly cookies to authenticate the request.

**Comment by enisdenjo** [comment, marked as answer]
The cookies should be automatically sent with the WebSocket upgrade request, are they not?

**Reply by christophediprima** [reply]
No they are not...

**Reply by enisdenjo** [reply]
Are you sure the cookie exists? Can you share a repro please? I just checked one of my apps that uses HttpOnly cookies and they're indeed included.

**Reply by enisdenjo** [reply]
Furthermore, the [WebSocket constructor](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/WebSocket) doesn't have any configuration options other than the URL and the subprotocols. Meaning, if the cookies are not included - it's a browser issue (which I doubt).

**Reply by christophediprima** [reply]
Not easy to provide you with a repro but sure, I can try if following details doesn't help.

One thing that may be specific to my use case is that the call is done from another sub-domain:

Cookie domain is set like so: `.domain.org`
The client app is hosted under: `https://client.domain.org`
The API is hosted under: `https://api.domain.org`

**Reply by christophediprima** [reply]
You were right! It is automatically sent with the WebSocket upgrade request! There were some issues on the server side! Sorry for bothering!
